src/workers/filesystems/infrastructure/SyncRemoteFileSystem.ts

```diff
@@ -0,0 +1,666 @@
+          fileSize: source.size,
+          source: source.stream,
+        });
+      });
+
+      const oldFileInCache = cache[name];
+
+      if (oldFileInCache) {
+        try {
+          const res = await httpRequest(
+            `${process.env.API_URL}/api/storage/folder/${oldFileInCache.parentId}/file/${oldFileInCache.id}`,
+            {
+              method: 'DELETE',
+              headers,
+            }
+          );
+          if (!res.ok) {
+            Logger.warn(
+              `Error trying to delete outdated remote file. res: ${await serializeRes(
+                res
+              )} fileInCache: ${JSON.stringify(oldFileInCache, null, 2)}`
+            );
+          }
+        } catch (e) {
+          const err = e as Error & { code: string };
+          Logger.warn(
+            `Error trying to delete outdated remote file. ${err.name} ${
+              err.code
+            } ${err.stack} fileInCache: ${JSON.stringify(
+              oldFileInCache,
+              null,
+              2
+            )}`
+          );
+        }
+      }
+
+      const encryptedName = crypt.encryptName(
+        baseNameWithoutExt,
+        folderIdOfTheNewFile.toString()
+      );
+
+      const modificationTime = getDateFromSeconds(modTimeInSeconds);
+
+      try {
+        const res = await httpRequest(
+          `${process.env.API_URL}/api/storage/file`,
+          {
+            headers,
+            method: 'POST',
+            body: JSON.stringify({
+              file: {
+                bucket,
+                encrypt_version: '03-aes',
+                fileId: uploadedFileId,
+                file_id: uploadedFileId,
+                folder_id: folderIdOfTheNewFile,
+                name: encryptedName,
+                plain_name: baseNameWithoutExt,
+                size,
+                type: fileType,
+                modificationTime,
+              },
+            }),
+          }
+        );
+        if (!res.ok) {
+          throw new ProcessError(
+            'BAD_RESPONSE',
+            createErrorDetails(
+              {},
+              'Creating file in drive server',
+              `res: ${await serializeRes(
+                res
+              )}, encryptedName: ${encryptedName}, modificationTime: ${modificationTime}`
+            )
+          );
+        }
+      } catch (err) {
+        await handleFetchError(
+          err,
+          'Creating file in drive server',
+          `encryptedName: ${encryptedName}, modificationTime: ${modificationTime}`
+        );
+      }
+    },
+
+    async pullFolder(name: string): Promise<void> {
+      const route = name.split('/');
+
+      const n = route.at(-1);
+
+      const parentFolderRoute = route.slice(0, route.length - 1).join('/');
+
+      const folderInCache = cache[parentFolderRoute];
+
+      const lastParentId =
+        folderInCache !== undefined ? folderInCache.id : baseFolderId;
+
+      httpRequest(`${process.env.API_URL}/api/storage/folder`, {
```

Comment:
same here, avoid the async-then mix

Reply:
I've actually deleted this part since the folders are created beforehand. On the new versions of the file systems (`SyncRemoteFileSystem` and `SyncLocalFileSystem`), the method pull file does only that, pull the file into the file system.